File: okio/src/integrationTest/java/okio/test/integration/ModuleTest.java
package okio.test.integration;

import okio.FileSystem;
import org.junit.Test;
import static org.junit.Assert.*;

public class ModuleTest {
  @Test
  public void test() {
    // test okio.test.integration is modular
    assertTrue(ModuleTest.class.getModule().isNamed());
    assertEquals(ModuleTest.class.getModule().getName(), "okio.test.integration");
    assertFalse(ModuleTest.class.getModule().getDescriptor().isAutomatic());
    // test okio is modular
    assertTrue(FileSystem.class.getModule().isNamed());
    assertEquals(FileSystem.class.getModule().getName(), "okio");
    assertFalse(FileSystem.class.getModule().getDescriptor().isAutomatic());
  }
}

File: okio/src/jvmIntegrationTest/kotlin/okio/test/integration/ModuleTest.kt
package okio.test.integration

import kotlin.test.Test
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class ModuleTest {
  @Test
  fun testModule() {
    // assert okio is modular
    val okioModule = ModuleLayer.boot().modules().single { it.name == "okio" }
    assertFalse(okioModule.descriptor.isAutomatic)
    assertTrue(okioModule.isExported("okio"))
    assertFalse(okioModule.isExported("okio.internal"))
  }
}
